### PR TITLE
Oral history request: admin page(s)

### PR DESCRIPTION
**Pull request by eddierubeiz**
This is the logging part of Ref #828:
**TODO:**
- [x] Staff should be able to download log in spreadsheet form, giving date ranges of requests to include.
- [x] Staff should be able to see "recent" delivered requests in browser.
- [x] Tests

**NOT PART OF THIS PR:**
Delivery of the actual OH content to the patron.

### CODE REVIEW

**Comment by jrochkind** [comment]
Set up a work in staging that is faked to have restricted-by-request media. https://staging-digital.sciencehistory.org/works/02870x01s

(Note Front Matter and Index PDF is from another OH, for demonstration purposes only)

File: spec/controllers/admin/oral_history_access_requests_spec.rb
Comment (jrochkind) [inline spec/controllers/admin/oral_history_access_requests_spec.rb:34]: let's actually "parse" the CSV returned. Instead of just testing that "Patron 8" is in the line SOMEWHERE, let's actually use ruby code to parse CSV, make sure it parses, and the various fields are in the right places. 

Also I just added a column for Work URL, so that can be included.

**Comment by jrochkind** [comment]
@MDiMeo  @lsberry16  The text on the form says:

> You are receiving a copy of this oral history and <strong>you are not permitted to reproduce it</strong> in any way without the permission of the Science History Institute.

But we're also giving them a "rights statement" that says "NON-COMMERCIAL USE PERMITTED", and links to a description saying: "no permission is required from the rights-holder(s) for non-commercial uses. For other uses you need to obtain permission from the rights-holder(s)."

I feel like we're giving them confusing mixed messages -- this stuff is confusing enough already. 

Thoughts?

(As a larger issue, I've thought for some time that for things where the Institute is the rights holder, we ought to be giving them a CC license, not just a rights statement, but that's a different larger issue and I could be wrong).

**Comment by jrochkind** [comment]
Code looks good @eddierubeiz, nice job. Made some UI tweaks, including to form that was part of previous UI. 

Made one comment about tests, which you are free to take or leave. Other than that, code is reviewed, but I think maybe it makes sense to demo and get approval from stakeholders before merging and closing issue? Can you take on that?

**Review by jrochkind — APPROVED** [review]

File: app/controllers/admin/oral_history_access_requests_controller.rb
Comment (jrochkind) [inline app/controllers/admin/oral_history_access_requests_controller.rb:8]: Would it make more sense to show most recent N requests instead of 3 months? Say most recent 50 or something? `.order(created_at: :desc).limit(50)`?  Not sure.

Comment (eddierubeiz) [reply app/controllers/admin/oral_history_access_requests_controller.rb:8]: It occurred to me. Will ask Lee.

**Comment by eddierubeiz** [comment]
Re: verbiage: this is above my pay grade, but I would suggest something like:

- Your receipt of an electronic copy of an oral history indicates your agreement to abide by U.S. copyright law **and the oral history's rights statement** with regard to the reproduction and dissemination of the oral history.
- Get rid of the part where it says you can't disseminate it; that's superseded by the above.
- Get rid of the "You are receiving a copy of this oral history" part -- that's superfluous.

**Comment by jrochkind** [comment]
Alternately it's possible the rights statement isn't right for these "free access no internet release" documents, and it could be changed to just "in copyright". 

I'm not opposed to providing additional guidance than just the phrase-and-link of our current rights statement; I am just concerned if doesn't seem to match!

**Comment by lsberry16** [comment]
Thanks, Eddie and Jonathan! Let me check in with Dave about the wording and get back to you.

Sent from my iPad


__________________________________

Lee Berry   (she, her, hers)
Curator of Oral Histories
Center for Oral History
t. +1.215.873.8251

Science History Institute
Chemistry • Engineering • Life Sciences
315 Chestnut Street • Philadelphia, PA 19106 • U.S.A.
Learn about the scientific discoveries that changed our world without leaving home at sciencehistory.org/learn<https://www.sciencehistory.org/learn>

On Oct 20, 2020, at 11:51 AM, Jonathan Rochkind <notifications@github.com> wrote:

﻿

Alternately it's possible the rights statement isn't right for these "free access no internet release" documents, and it could be changed to just "in copyright".

I'm not opposed to providing additional guidance than just the phrase-and-link of our current rights statement; I am just concerned if doesn't seem to match!

File: spec/controllers/admin/oral_history_access_requests_spec.rb
Comment (eddierubeiz) [reply spec/controllers/admin/oral_history_access_requests_spec.rb]: Thanks for adding the work URL (duh!). CSV is parsed now.

**Comment by eddierubeiz** [comment]
Include the Oral History number as well in both the table in admin and the downloaded spreadsheet.
The Oral History number is listed in the work metadata under the label "External ID".

**Comment by eddierubeiz** [comment]
Confirmation message should read "Links to the files are being sent" rather than "Files are being sent"

File: app/controllers/admin/oral_history_access_requests_controller.rb
Comment (eddierubeiz) [reply app/controllers/admin/oral_history_access_requests_controller.rb:8]: No, past 3 months is fine, said Lee.

Comment (jrochkind) [reply app/controllers/admin/oral_history_access_requests_controller.rb:8]: I'm worried past 3 months would be TOO MANY lines, too big a page, performance problems. Since it's unbounded as to actual number, it just depends on how many requests there have been. Is why I would have made it a fixed number. But we can just deal with that if it becomes a problem, I guess.

**Pull request merged**
